packages/clerk-js/src/core/resources/Session.ts

```diff
@@ -118,7 +118,9 @@ export class Session extends BaseResource implements SessionResource {
   // e.g. session id is 'sess_abc12345' and jwt template name is 'haris'
   // The session token ID will be 'sess_abc12345' and the jwt template token ID will be 'sess_abc12345-haris'
   #getCacheId(template?: string, organizationId?: string) {
-    return [this.id, template, organizationId, this.updatedAt.getTime()].filter(Boolean).join('-');
+    const resolvedOrganizationId =
+      typeof organizationId === 'undefined' ? this.lastActiveOrganizationId : organizationId;
+    return [this.id, template, resolvedOrganizationId, this.updatedAt.getTime()].filter(Boolean).join('-');
```

Comment:
additional bug: when the token cache is hydrated on init, the cache key didn't include the organization ID.